Subject: [PATCH] [BZ 168517] dbginfo.sh: Extend data collection
From: Sa Liu <saliu@de.ibm.com>

Description:    dbginfo.sh: Extend data collection 
Symptom:        This update covers various symptoms on dbginfo.sh data
                collection:
                - There is no data collected for docker.
                - ps command does not show threads infomation.
                - There is no run queue statistics and scheduler data. 
                - z/VM commands do not show multithread, protect
                  settings and SSI status.
Problem:        Following problems exist:
                - No analysis is possible for docker data.
                - Missing thread information.
                - Missing run queue statistics and scheduler data.
                - Missing z/VM information.
Solution:       - Extend the data collection to collect docker data
                - Change ps command to show threads informaton
                - Add commands to display run queue statistics and 
                  scheduler data.
                - Add z/VM commands to show multithread, protect settings
                  and SSI status.
Reproduction: Run this script and verify the output
Upstream-ID:  -
Problem-ID:   168517

Signed-off-by: Sa Liu <saliu@de.ibm.com>
---
 scripts/dbginfo.sh |    8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

--- a/scripts/dbginfo.sh
+++ b/scripts/dbginfo.sh
@@ -375,7 +375,8 @@ CMDS="uname -a\
   :runlevel\
   :iptables -L\
   :ulimit -a\
-  :ps -eo pid,tid,nlwp,policy,user,tname,ni,pri,psr,sgi_p,stat,wchan,start_time,time,pcpu,pmem,vsize,size,rss,share,command\
+  :ps -emo pid,tid,nlwp,policy,user,tname,ni,pri,psr,sgi_p,stat,wchan,start_time,time,pcpu,pmem,vsize,size,rss,share,command\
+  :ps -eHo pid,tid,nlwp,policy,user,tname,ni,pri,psr,sgi_p,stat,wchan,start_time,time,pcpu,pmem,vsize,size,rss,share,command\
   :ps axX\
   :dmesg -s 1048576\
   :last\
@@ -470,6 +471,8 @@ VM_CMDS="q userid\
   :q privclass\
   :q cplevel\
   :q cpservice\
+  :q cpprot user\
+  :q specex\
   :q ssi\
   :q cpus\
   :q srm\
@@ -508,6 +511,7 @@ VM_CMDS="q userid\
   :q cache\
   :q nic\
   :q pav\
+  :q proc\
   :q proc topology\
   :q mt\
   :q qioass\
@@ -831,7 +835,7 @@ post_processing() {
     local tmp_file
     local file_name
 
-    pr_syslog_stdout "11 of ${COLLECTION_COUNT}: Postprocessing"
+    pr_syslog_stdout "${COLLECTION_COUNT} of ${COLLECTION_COUNT}: Postprocessing"
 
     find "${WORKPATH}etc/libvirt/qemu/" -maxdepth 1 -name "*.xml" 2>/dev/null | while IFS= read -r file_name; do
 	file_mtime_epoche=$(stat --format=%Y "${file_name}")
